test/test_cases/run_2d_moist_bubble.jl

```diff
@@ -0,0 +1,161 @@
+        set!(simulation, :base, ρ = ρ, ρuh = ρuh, ρw = ρw)
+        set!(simulation, :thermodynamics, ρe_tot = ρe_tot)
+        set!(simulation, :moisture, ρq_tot = ρq_tot)
+
+        # initial state
+        u = simulation.integrator.u
+        ∫ρ_0 = sum(u.base.ρ)
+        ∫ρe_tot_0 = sum(u.thermodynamics.ρe_tot)
+        ∫ρq_tot_0 = sum(u.moisture.ρq_tot)
+
+        # 1 step
+        step!(simulation)
+        u_end = simulation.integrator.u
+        ∫ρ_e = sum(u.base.ρ)
+        ∫ρe_tot_e = sum(u.thermodynamics.ρe_tot)
+        ∫ρq_tot_e = sum(u.moisture.ρq_tot)
+
+        # perform regression check
+        if FT == Float32
+            current_min = 236505.89f0
+            current_max = 253207.83f0
+        elseif FT == Float64
+            current_min = 236505.86276988868
+            current_max = 253207.81761817695
+        else
+            error("Float type $FT not tested.")
+        end
+
+        @test minimum(parent(u_end.thermodynamics.ρe_tot)) ≈ current_min atol =
+            0.01
+        @test maximum(parent(u_end.thermodynamics.ρe_tot)) ≈ current_max atol =
+            0.01
+
+        # conservation check
+        Δρ = (∫ρ_e - ∫ρ_0) ./ ∫ρ_0 * 100
+        Δρe_tot = (∫ρe_tot_e - ∫ρe_tot_0) ./ ∫ρe_tot_0 * 100
+        Δρq_tot = (∫ρq_tot_e - ∫ρq_tot_0) ./ ∫ρq_tot_0 * 100
+
+        if FT == Float32
+            @test abs(Δρ) < 3e-5
+            @test abs(Δρe_tot) < 4e-5
+            @test abs(Δρq_tot) < 1e-3
+        else
+            @test abs(Δρ) < 1e-12
+            @test abs(Δρe_tot) < 1e-5
+            @test abs(Δρq_tot) < 1e-3
+        end
+
+    elseif test_mode == :validation
+        # path = joinpath(@__DIR__, first(split(basename(@__FILE__), ".jl")))
+        # mkpath(path)
+        # cb_jld2 = JLD2Output(model, path, "moist_bubble", dt)
+        # cb_set = CallbackSet(generate_callback(cb_jld2))
+        simulation = Simulation(model, stepper, dt = dt, tspan = (0.0, 500.0))#, callbacks = cb_set,)
+        @unpack ρ, ρuh, ρw, ρe_tot, ρq_tot =
+            init_2d_moist_bubble(FT, params, thermovar = :ρe_tot)
+        set!(simulation, :base, ρ = ρ, ρuh = ρuh, ρw = ρw)
+        set!(simulation, :thermodynamics, ρe_tot = ρe_tot)
+        set!(simulation, :moisture, ρq_tot = ρq_tot)
+        run!(simulation)
+        u_end = simulation.integrator.u
+
+        # post-processing
+        ENV["GKSwstype"] = "nul"
+        Plots.GRBackend()
+
+        # make output directory
+        path = joinpath(@__DIR__, first(split(basename(@__FILE__), ".jl")))
+        mkpath(path)
+
+        foi = Plots.plot(u_end.moisture.ρq_tot ./ u_end.base.ρ)
+        Plots.png(foi, joinpath(path, "moist_rising_bubble_2d_FT_$FT"))
+
+        @test true # check is visual
+    else
+        throw(ArgumentError("$test_mode incompatible with test case."))
+    end
+
+    nothing
+end
+
+@testset "2D moist rising bubble" begin
+    for FT in (Float32, Float64)
+        run_2d_moist_bubble(FT)
+    end
+end
```

Comment:
We should probably make `FT` a CLI, like the other tests to encourage better CI parallelism

Reply:
Yes, this is a common issue for all test cases (except Ekman layer). Let's move it out in a separate PR #249 .